### PR TITLE
MULE-14939: Document OAuth2 Provider module

### PR DESCRIPTION
**Pull request by lucianoRM**

### CODE REVIEW

File: mule4-user-guide/v/4.1/migration-oauth2-provider.adoc [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+
+=== Revoke Token
+
+The operation remains the same.
+
+== Example
+
+Here is an example of the same application configured in Mule 3 And Mule 4.
+
+The application has an OAuth2 Provider that grants tokens and a flow that listens to http requests and has a token validation before processing some logic.
```
Comment (aiannucci) [inline mule4-user-guide/v/4.1/migration-oauth2-provider.adoc]: http -> HTTP

File: mule4-user-guide/v/4.1/migration-oauth2-provider.adoc [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+
+Keep in mind that the Mule 4 configuration is using the Spring Module and the Object Store Connector.
+
+In both cases the application has been splitted in multiple files.
```
Comment (aiannucci) [inline mule4-user-guide/v/4.1/migration-oauth2-provider.adoc]: splitted -> split

Comment (aiannucci) [reply mule4-user-guide/v/4.1/migration-oauth2-provider.adoc]: in -> into

File: mule4-user-guide/v/4.1/migration-oauth2-provider.adoc [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+
+For Mule 3 there are 2: One for common configuraion and other other the actual OAuth2 Provider configuration. +
```
Comment (aiannucci) [inline mule4-user-guide/v/4.1/migration-oauth2-provider.adoc]: configuaion -> configuration
other other -> another one for

File: mule4-user-guide/v/4.1/migration-oauth2-provider.adoc [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+==== Bean Configuration
+[source, xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:ss="http://www.springframework.org/schema/security"
+       xsi:schemaLocation="
+       http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-4.0.xsd
+       http://www.springframework.org/schema/security http://www.springframework.org/schema/security/spring-security-4.0.xsd">
```
Comment (aiannucci) [inline mule4-user-guide/v/4.1/migration-oauth2-provider.adoc]: Remove schema versions, this depends on the Spring version they decide to use.

File: connectors/v/latest/oauth2-provider-documentation-reference.adoc [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+| Authorization Config | Authorization Config | Configuration related to authorization code handling | {nbsp} | {nbsp}
+| Token Config | Token Config | Configuration related to token handling | {nbsp} | {nbsp}
+|======================
+
+
+=== Security Providers
+
+As seen in the <<general-configuration>>, two security providers should be defined in the application to be later referenced by the OAuth2 Configuration element.
+
+One way of doing this is by using the Spring Framework, defining both security providers and then using the link:spring-module.adoc[Spring Module] to add them to the Mule Security Manager as so:
```
[the remark this answers is not in the document]
Comment (lucianoRM) [reply connectors/v/latest/oauth2-provider-documentation-reference.adoc]: There is an example to the spring module. You mean add one to spring itself?

File: connectors/v/latest/oauth2-provider-documentation-reference.adoc [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+| Secret | String | The client's secret (password) used for authentication. | {nbsp} | Only if the client type is CONFIDENTIAL
+| Redirect Uris | Expression | An expression that resolves to a list of redirect uris used for when the client makes requests to the OAuth Provider | Empty List | {nbsp}
+| Authorized Grant Types | Expression | An expression that resolves to a list of the authorized grant types that the client can use to request a token. Valid values are: AUTHORIZATION_CODE,REFRESH_TOKEN, TOKEN, PASSWORD, CLIENT_CREDENTIALS. | Empty List | {nbsp}
+| Scopes | Expression |  An expression that resolves to a list of supported scopes by the client. If none provider, the default scopes of the <<general-configuration>> will be used | Empty List | {nbsp}
+| Fail if Present | Boolean | Defines what to do if the a client with the same Id is already registered. If true, an error will be raised. Otherwise, the client will be updated | false | {nbsp}
+|======================
+
+==== Raises
+* OAUTH2-PROVIDER:CLIENT_ALREADY_EXISTS(OAUTH_SERVER_SECURITY)
```
[the remark this answers is not in the document]
Comment (lucianoRM) [reply connectors/v/latest/oauth2-provider-documentation-reference.adoc]: Yes

File: connectors/v/latest/oauth2-provider-documentation-reference.adoc [hunk 7]
```diff
@@ -0,0 +1,365 @@
...
+[[oauth2-provider-reference]]
+== OAuth2 Provider Module Documentation Reference
+
+The OAuth2 Provider Module allows a Mule Application to be configured as an Authentication Manager in an OAuth2 Dance.
+With this role, the application will be able to authenticate previously registered clients, grant tokens and validate them or register and delete clients, all during the execution of a flow.
+
+
+[WARNING]
+The following documentation assumes a basic knowledge of the OAuth2 authorization protocol. If you are not familiar with the it, the RFC specification can be found in: link:https://tools.ietf.org/html/rfc6749[RFC6749] +
+Also, if the app should behave as the *Client in the OAuth2 Dance*, then the OAuth Module should be used.
```
Comment (aiannucci) [inline connectors/v/latest/oauth2-provider-documentation-reference.adoc]: Link to OAuth module docs

**Pull request merged**